### ISSUE TITLE
IME input problem.

### ISSUE THREAD

**Issue by urjskhzk** [ocornut/imgui#4642]
**Version/Branch of Dear ImGui:**

Version: 1.84.2
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: win32 dx9
Compiler: vs2019
Operating System: Windows 10

**My Issue/Question:**

This is a case when WM_CHAR and WM_LBUTTONDOWN events occur at the same time.

io.MouseDown[0]: true
io.InputQueueCharacters: Size 1, Data 44032 '가'

1. Good

```
static bool b = false;
static char buf[256] = { 0, };
ImGui::Begin("Hello, world!");
ImGui::InputText("InputText", buf, _countof(buf));
ImGui::Checkbox("Checkbox", &b);
ImGui::End();
```

![1](https://user-images.githubusercontent.com/36331749/136966656-ae5b9f83-4a76-4d01-b979-230209629acb.gif)

2. Bad

```
static bool b = false;
static char buf[256] = { 0, };
ImGui::Begin("Hello, world!");
ImGui::Checkbox("Checkbox", &b);
ImGui::InputText("InputText", buf, _countof(buf));
ImGui::End();
```

![2](https://user-images.githubusercontent.com/36331749/136966684-2112d3e8-a3ec-4fd6-8ea9-2647bd7a755a.gif)

3. A is Good / B is Bad

```
static char buf1[36] = { 0, };
static char buf2[36] = { 0, };
ImGui::Begin("Hello, world!");
ImGui::InputText("A", buf1, _countof(buf1));
ImGui::InputText("B", buf2, _countof(buf2));
ImGui::End();
```

![3](https://user-images.githubusercontent.com/36331749/136984965-5c03ec99-6049-4e62-9e60-49c02c98d335.gif)

**Screenshots/Video**

**Standalone, minimal, complete and verifiable example:**

**Comment by ocornut** [comment]
This is not an IME specific problem but an IME triggering both events simultaneously would exacerbate the problem.

If you use a custom input queue like https://gist.github.com/ocornut/8417344f3506790304742b07887adf9f you can problably work around this problem. We are now working (well, thedmd is) on an official version of this to be in master.

**Comment by urjskhzk** [comment]
This is a situation that occurs frequently in the Korean input proces.
I don't know about Chinese and Japanese input.
![4](https://user-images.githubusercontent.com/36331749/137085720-4122124e-def6-44f3-bef7-cb135c93dbcd.png)
s.

**Comment by LunaTheFoxgirl** [comment]
Can confirm that the problem also happens with Japanese IME and in some cases (due to the SDL2 backend Inochi Creator is using) can cause crashes due to cut-off UTF8 sequences.)
Though I think it's a bit of a multi-fold problem in our case.

**Comment by ocornut** [comment]
Hello,

Was looking into this day.
AFAIK this should be fixed by 1.87 event queue with a conforming backend (#4921) since text input trickles when preceded by a mouse down on same frame.

However, I can't repro (with `io.ConfigInputTrickleEventQueue = false;`) before with my IME showing a suggestion window, the first click is inhibited by win32 popup logic and I don't get the mouse down. However pretty sure it would now handle it properly with 1.87, but let me know if it doesn't!

**Comment by ocornut** [comment]
PS: In the very last commit in Tools>DebugLog I have added a `[X] IO` checkbox allowing to visualize input events and how they are trickled.
